src/main/java/com/lightbend/akka/jpa/sample/User.java

```diff
@@ -0,0 +1,73 @@
+package com.lightbend.akka.jpa.sample;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "USERS")
+public class User {
+  // primary key
+  @Id
+  @GeneratedValue(generator = "increment")
+  @GenericGenerator(name = "increment", strategy = "increment")
+  private long id;
+
+  // optimistic locking
+  @Version
+  private long version;
+
+  private final String name;
```

Comment:
I didn't even know Hibernate could work with classes that use final fields. Nice!